### PR TITLE
Simplified setup.py code, reduced code duplication

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Simplified setup.py code, reduced code duplication

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: features/setup.py [hunk 1]
```diff
@@ -1,82 +1,39 @@
...
+version = ag.load_version_file()
+version = ag.update_version(version)
+
+submodule = 'features'
 requirements = [
-    'numpy==1.19.5',
-    'pandas>=1.0.0,<2.0',
-    'scikit-learn>=0.22.0,<0.25',
-    f'autogluon.core=={version}'
+    'numpy',
```
Comment (gradientsky) [inline features/setup.py:20]: nit: to avoid confusion, please add a comment that pinned versions will be added in `get_dependency_version_ranges` call

**Review by gradientsky — APPROVED** [review]

**Review by gradientsky — APPROVED** [review]

**Comment by szha** [comment]
Job PR-934-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-934/1/index.html

File: features/setup.py (hunk 1 above)
Comment (Innixma) [reply features/setup.py:20]: added

**Comment by szha** [comment]
Job PR-934-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-934/2/index.html

**Comment by szha** [comment]
Job PR-934-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-934/4/index.html

**Pull request merged**